rmf_robot_sim_gz_plugins/src/TeleportDispenser.cpp

```diff
@@ -282,28 +486,7 @@ void TeleportDispenserPlugin::PreUpdate(const UpdateInfo& info,
     tried_fill_dispenser = true;
   }
 
-  std::function<void(FleetStateIt,
-    std::vector<SimEntity>&)> fill_robot_list_cb =
-    std::bind(&TeleportDispenserPlugin::fill_robot_list, this,
-      std::ref(ecm), std::placeholders::_1, std::placeholders::_2);
-
-  std::function<SimEntity(const std::vector<SimEntity>&,
-    bool&)> find_nearest_model_cb =
-    std::bind(&TeleportDispenserPlugin::find_nearest_model, this,
-      std::ref(ecm), std::placeholders::_1, std::placeholders::_2);
-
-  std::function<void(const SimEntity&)> place_on_entity_cb =
-    std::bind(&TeleportDispenserPlugin::place_on_entity, this,
-      std::ref(ecm), std::placeholders::_1, _item_en);
-
-  std::function<bool(void)> check_filled_cb = [&]()
-  {
-    return _dispenser_vicinity_box.Contains(
-      ecm.Component<components::Pose>(_item_en)->Data().Pos());
-  };
-
-  _dispenser_common->on_update(fill_robot_list_cb, find_nearest_model_cb,
-    place_on_entity_cb, check_filled_cb);
+  on_update(ecm);
```

Comment:
I can see where this came from (porting the `on_update` function) but I don't personally find it very useful. The function name doesn't say what it does and it encapsulates a lot of logic, I would suggest either splitting up the logic into different functions or just bringing the code in the `PreUpdate` call